rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_statement_wrapper.cpp

```diff
@@ -65,6 +65,28 @@ std::shared_ptr<SqliteStatementWrapper> SqliteStatementWrapper::execute_and_rese
   return reset();
 }
 
+std::shared_ptr<SqliteStatementWrapper> SqliteStatementWrapper::execute_and_check_value(
+  bool & check_value)
+{
+  int return_code = sqlite3_step(statement_);
+  if (!is_query_ok(return_code)) {
+    std::stringstream errmsg;
+    errmsg << "Error when checking value of SQL statement. SQLite error (" <<
+      return_code << "): " << sqlite3_errstr(return_code);
+
+    throw SqliteException{errmsg.str()};
+  }
+
+  check_value = true;
+  bool no_result = return_code == SQLITE_DONE || sqlite3_column_count(statement_) == 0;
+
+  if (no_result || sqlite3_column_type(statement_, 0) == SQLITE_NULL) {
+    // No result or result is null means that no such pragma exists
+    check_value = false;
+  }
+  return reset();
+}
+
```

Comment:
I think this should be a function overlay of `execute_and_reset`. This makes the behavior clearer, but also avoids code duplication.

Reply:
Will make this one an overload of the other function (you meant overload, right? I am not familiar with the term overlay in this context). 

As for testing, the first thing that comes to mind is to look for warnings in the output. Currently I added WARN logs, e.g. ones that show up when the pragma is an unrecognized setting: 

[WARN] [1603730319.797767516] [rosbag2_storage]: PRAGMA setting not recognized by sqlite, not applied: bad_pragma_name
[WARN] [1603730319.797800819] [rosbag2_storage]: PRAGMA setting not recognized by sqlite, not applied: worse_pragma_name

Would that be ok?

Reply:
yes, overload!

Warning parsing seems very tedious for me and most likely indicates a misleading API. Is there a chance we can throw when pragmas are wrongly set?

Reply:
Of course we can - and I agree this is better than ignoring and warning. I will implement that!

Reply:
since the change is actually only adding an optional check, I added an optional parameter instead of an overload